applications/OptimizationApplication/python_scripts/utilities/response_utilities.py

```diff
@@ -0,0 +1,246 @@
+import re
+from enum import Enum
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.responses.response_function import ResponseFunction
+from KratosMultiphysics.OptimizationApplication.responses.evaluation_response_function import EvaluationResponseFunction
+from KratosMultiphysics.OptimizationApplication.responses.response_function import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.component_data_view import ComponentDataView
+
+class BinaryOperator(str, Enum):
+    POWER    = "^"
+    DIVIDE   = "/"
+    MULTIPLY = "*"
+    ADD      = "+"
+    SUBTRACT = "-"
+
+BinaryOperatorValues = [operator.value for operator in BinaryOperator]
+
+BinaryOperatorValuesMap = dict([(operator.value, operator) for operator in BinaryOperator])
+
+def GetClosingBracketIndex(expression: str) -> int:
+    # this function assumes the starting point is without the opening bracket.
+    # it will give the position of the respective closing bracket
+    index = 0
+    brackets_counter = 1
+    while index < len(expression):
+        brackets_counter += (expression[index] == "(") + (expression[index] == ")") * -1
+        if brackets_counter == 0:
+            return index
+        index += 1
+
+    raise RuntimeError(f"The closing bracket not found for expression ({expression}")
+
+def GetFunction(function_name: str, optimization_problem: OptimizationProblem, *args) -> ResponseFunction:
+    # import functions
+    from KratosMultiphysics.OptimizationApplication.responses.log_response_function import LogResponseFunction
+
+    functions_map: 'dict[str, type[ResponseFunction]]' = {
+        "log": LogResponseFunction
+    }
+    if function_name in functions_map.keys():
+        return functions_map[function_name](*args, optimization_problem)
+    else:
+        raise RuntimeError(f"Undefined \"{function_name}\" function. Followings are supported function names:\n\t" + "\n\t".join(functions_map.keys()))
+
+def EvaluateValue(response_function: ResponseFunction, optimization_problem: OptimizationProblem) -> float:
+    response_data = ComponentDataView("evaluated_responses", optimization_problem).GetUnBufferedData()
+
+    if not response_data.HasValue(f"values/{response_function.GetName()}"):
```

Comment:
I am wondering if it is safe to check only in the data because we usually check based on the given x ...